snuba/datasets/events.py

```diff
@@ -345,6 +345,26 @@ def column_expr(
         )
         if processed_column:
             # If processed_column is None, this was not a tag/context expression
+
+            # This conversion must not be ported to the errors dataset. We should
+            # not support promoting tags/contexts with boolean values. There is
+            # no way to convert them back consistentyl to the value provided by
+            # the client when the event is ingested, in all ways to access
+            # tags/contexts. Once the errors dataset is in use, we will not have
+            # boolean promoted tags/contexts so this constraint will be easy to enforce.
```

Comment:
Is this (type consistency across promotion) something that we can enforce in the dataset definition when/as we introduce entities?

Reply:
No, the issue is orthogonal. The consistency issue is that, when we ingest an event and we promote a tag/context with boolean value we convert "true", "True", "Yes", "1" to true independently of the original value. The original value is lost so we cannot return it back later on.
If the client provided "1" we will return "True" from the promoted column, we don't know anymore that the original value was "1"